scripts/west_commands/runners/nrf_common.py

```diff
@@ -357,7 +357,8 @@ def program_hex(self):
 
             if self.erase:
                 self.exec_op('erase', core='Application', kind='all')
-                self.exec_op('erase', core='Network', kind='all')
+                if not self.build_conf.get('CONFIG_NRF_IRON'):
+                    self.exec_op('erase', core='Network', kind='all')
```

Comment:
so if a script needs this then it should be soc level, not board level, and this should probably throw an error in the compliance check (if it checked script files)

Reply:
actually I don't get this, if this is with e.g. MCUboot then a program operation for an application core image should program/erase **just** the application core, likewise a program operation for the network core should program/erase **just** the network core, the handling here doesn't seem to make sense

Reply:
I updated it so it performs a "global" erase operation for the NRF_IRON case, since there exists no "domain erase" operation for that variant.